### DISCUSSION TITLE
citation feature?

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#2763] [Feature Requests & Ideas]
I guess it will be just like link but with superscript in richtext.

Like the references panel in wikipedia: https://en.wikipedia.org/wiki/Hedera#cite_note-WT-6

**Comment by blueMountain007** [comment]
The same scenario also requires this

**Comment by blueMountain007** [comment]
@denolfe  Any ideas? Thanks!

**Comment by DanRibbens** [comment]
I would try to do this using the [custom link fields](https://payloadcms.com/docs/fields/rich-text#building-custom-elements-and-leaves:~:text=RichText%20link%20with%20custom%20fields) feature and add a field `type: 'checkbox'` for "citation". Then in your frontend you can loop all the "citation" links from your JSON and render them out in the page footer.

Alternatively, you could extend your richtext editor to give it it's own toolbar action from links. This would be nicer UX, but then you have to copy and then maintain your own implementation that is essentially a copy of the built-in richtext `link` feature that Payload already has.

Ideally your frontend app is also resposible for maintaing the numbering of the citations so that editors don't have cascade their updates throughout a large document if they add/remove a citation early in a document. 

What do you think?

**Comment by blueMountain007** [comment]
Thanks for your idea, very beautiful idea!
